contracts/REXTwoWayMarket.sol

```diff
@@ -372,138 +234,157 @@ contract REXTwoWayMarket is REXMarket {
             _timestamp,
             uint256(uint96(_flowRateMain)),
             // Select the correct lastDistributionAt for this _superToken
-            _getLastDistributionAt(_superToken)
+            market.lastDistributionAt
         );
         _cbdata = abi.encode(_uinvestAmount, int256(_flowRateMain));
     }
 
+
+    // Src: Charm Finance, Unlicense
+    // https://github.com/charmfinance/alpha-vaults-contracts/blob/07db2b213315eea8182427be4ea51219003b8c1a/contracts/AlphaStrategy.sol#L136
+    // Modified to return a price in 1e6 decimals
+    function getTwap() public view returns (uint _price) {
```

Comment:
@shreyaspapi this implementation of the oracle interface works for USDC>>ETH, I am working on breaking up that large formula.

Unfortately I could get this to work on a USDC>>rexSHIRT Uniswap LP, I think it has to do with the way I do the math